### ISSUE TITLE
redownload of deleted http upload messages still fails with 2.0.0 (f-droid)

### ISSUE THREAD

**Issue by jannic** [siacs/Conversations#2958]
#### General information

* **Version:** 2.0.0
* **Device:** Moto G 2015
* **Android Version:** 7.1.2 (LineageOS 14.1)
* **Server name:** self hosted
* **Server software:** ejabberd 18.01 (debian package 18.01-2~bpo9+1)
* **Installed server modules:** all necessary so Conversations shows 'avaliable' for all features
* **Conversations source:** F-Droid

#### Steps to reproduce

1. receive image sent using HTTP upload (OMEMO or unencrypted both show same behaviour)
2. delete image
3. select 'download image'

#### Expected result

Image is downloaded, again

#### Actual result

Message "You are not connected. Try again later"

#### Debug output

not available at the moment, but can retry and gather debug output if necessary

**Comment by iNPUTmice** [comment]
fixed in 2.1

**Comment by jannic** [comment]
Verified, the issue is fixed.

BTW, I traced it down to treatAsDownloadable() not handling the '|'-separated body parts containing file size etc., which in turn became part of the url to be checked for validity.
As treatAsDownloadable() is still used by other code, maybe there are bugs left at those points? Shouldn't treatAsDownloadable() handle body parts the same way getFileParams() does?
(Of course, if treatAsDownloadable() is only called before those parts are added, all is fine. Didn't verify that, though.)

PS: Thanks for adding the "not encrypted for this device" message, very helpful!
